### PR TITLE
Adjust log verbosity

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Following up on harmonizing log level definition.

### Motivation

Have better agent logs?

### CODE REVIEW

File: pkg/collector/scheduler.go [hunk 1]
```diff
@@ -60,7 +60,6 @@ func InitCheckScheduler(collector *Collector) *CheckScheduler {
 func (s *CheckScheduler) Schedule(configs []integration.Config) {
 	checks := s.GetChecksFromConfigs(configs, true)
 	for _, c := range checks {
-		log.Infof("Scheduling check %s", c)
```
Comment (mfpierre) [inline pkg/collector/scheduler.go]: redundant with https://github.com/DataDog/datadog-agent/blob/3a9bf7ca3136779799e0fc85f33afe4d9a1a253f/pkg/collector/scheduler/scheduler.go#L76

File: pkg/trace/agent/run.go [hunk 2]
```diff
@@ -94,22 +94,22 @@ func Run(ctx context.Context) {
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	agnt := NewAgent(ctx, cfg)
-	log.Infof("trace-agent running on host %s", cfg.Hostname)
+	log.Infof("Trace-agent running on host %s", cfg.Hostname)
```
Comment (gbbr) [inline pkg/trace/agent/run.go]: I think this case should stay lower cased. The binary name is `trace-agent`.

File: pkg/trace/config/apply.go [hunk 3]
```diff
@@ -275,7 +275,7 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		for key, rate := range rateBySpan {
 			serviceName, operationName, err := parseServiceAndOp(key)
 			if err != nil {
-				log.Errorf("error parsing names: %v", err)
+				log.Errorf("Rrror parsing names: %v", err)
```
Comment (gbbr) [inline pkg/trace/config/apply.go]: s/Rrror/Error

File: pkg/trace/agent/sampler.go [hunk 4]
```diff
@@ -114,7 +114,7 @@ func (s *Sampler) logStats() {
 
 			switch state := state.(type) {
 			case sampler.InternalState:
-				log.Debugf("%s: inTPS: %f, outTPS: %f, maxTPS: %f, offset: %f, slope: %f, cardinality: %d",
+				log.Tracef("%s: inTPS: %f, outTPS: %f, maxTPS: %f, offset: %f, slope: %f, cardinality: %d",
```
Comment (gbbr) [inline pkg/trace/agent/sampler.go]: @AlexJF this is fine, right?

File: pkg/trace/agent/run.go (hunk 2 above)
Comment (mfpierre) [reply pkg/trace/agent/run.go]: Maybe it should be change to "Trace agent" then 🤔

File: pkg/trace/agent/sampler.go (hunk 4 above)
Comment (mfpierre) [reply pkg/trace/agent/sampler.go]: Note that there's a debug log above with more generic info
`log.Debugf("%s: flushed %d sampled traces out of %d", engineType, keptTraceCount, totalTraceCount)`

File: pkg/trace/agent/run.go (hunk 2 above)
Comment (gbbr) [reply pkg/trace/agent/run.go]: I think I'd be fine with either. If we use the former, the most correct would be to simply use `os.Args[0]` which is the binary name. I'm ok with either. Up to you. 🙇

File: pkg/dogstatsd/server.go [hunk 5]
```diff
@@ -249,16 +249,12 @@ func (s *Server) parsePacket(packet *listeners.Packet, metricSamples []*metrics.
 	extraTags := s.extraTags
 
 	if packet.Origin != listeners.NoOrigin {
-		log.Tracef("Dogstatsd receive from %s: %s", packet.Origin, packet.Contents)
```
Comment (hush-hush) [inline pkg/dogstatsd/server.go]: Why remove the Tracef here ? Not useful ?

**Review by hush-hush — DISMISSED** [review]
Thanks for this !

Added one question but feel free to ignore

File: pkg/dogstatsd/server.go (hunk 5 above)
Comment (mfpierre) [reply pkg/dogstatsd/server.go]: Was useful in when the feature was being develop, but it's too verbose, when you have significant DSD traffic the logger can't keep up and the memory of the agent sky rockets.

File: pkg/trace/agent/run.go [hunk 6]
```diff
@@ -66,7 +66,7 @@ func Run(ctx context.Context) {
 	if flags.CPUProfile != "" {
 		f, err := os.Create(flags.CPUProfile)
 		if err != nil {
-			log.Critical(err)
+			log.Error(err)
```
Comment (gbbr) [inline pkg/trace/agent/run.go]: Why did you change this? If the flag is on for CPU profiling, this is surely critical isn't it?

File: pkg/trace/agent/run.go [hunk 7]
```diff
@@ -94,22 +94,22 @@ func Run(ctx context.Context) {
...
-	log.Infof("trace-agent running on host %s", cfg.Hostname)
+	log.Infof("Trace agent running on host %s", cfg.Hostname)
 	agnt.Run()
 
 	// collect memory profile
 	if flags.MemProfile != "" {
 		f, err := os.Create(flags.MemProfile)
 		if err != nil {
-			log.Critical("could not create memory profile: ", err)
+			log.Error("Could not create memory profile: ", err)
```
Comment (gbbr) [inline pkg/trace/agent/run.go]: Same question here.

File: pkg/trace/agent/run.go [hunk 8]
```diff
@@ -94,22 +94,22 @@ func Run(ctx context.Context) {
...
 		}
 
 		// get up-to-date statistics
 		runtime.GC()
 		// Not using WriteHeapProfile but instead calling WriteTo to
 		// make sure we pass debug=1 and resolve pointers to names.
 		if err := pprof.Lookup("heap").WriteTo(f, 1); err != nil {
-			log.Critical("could not write memory profile: ", err)
+			log.Error("Could not write memory profile: ", err)
```
Comment (gbbr) [inline pkg/trace/agent/run.go]: ...and here.

File: pkg/trace/agent/sampler.go (hunk 4 above)
Comment (gbbr) [reply pkg/trace/agent/sampler.go]: I guess this should be fine. I'm not aware of anyone actually making use of this.

**Review by gbbr — COMMENTED** [review]
Mostly looks good to me. A bit unsure about some level changes though.

File: pkg/trace/agent/run.go (hunk 6 above)
Comment (mfpierre) [reply pkg/trace/agent/run.go]: `Critical` errors should force the shutdown of the agent (as per the RFC)

File: pkg/trace/agent/run.go (hunk 7 above)
Comment (mfpierre) [reply pkg/trace/agent/run.go]: ditto

File: pkg/trace/agent/run.go (hunk 8 above)
Comment (mfpierre) [reply pkg/trace/agent/run.go]: ditto

**Review by gbbr — DISMISSED** [review]
